superset/views/core.py

```diff
@@ -3102,13 +3102,27 @@ def columns(self) -> str:
     @has_access_api
     @expose("/geocoding/geocode", methods=["POST"])
     def geocode(self) -> Response:
-        # TODO this has to be removed and replaced with a call to the get_data_from_table method
+        table_name = request.get("datasource", "")
+        columns = [request.form["streetColumn"], request.form["cityColumn"], request.form["countryColumn"]]
+        lat_column = request.get("latitudeColumnName", "lat")
+        lon_column = request.get("longitudeColumnName", "lon")
+
+        self._load_data_from_columns(table_name, columns)
+
+        try:
+            self._add_lat_lon_columns(table_name, lat_column, lon_column)
+        except ValueError as e:
+            return json_error_response(e.args[0], status=400)
+        except Exception as e:
+            message = "Error happens while create new columns for latitude and longitude"
+            return json_error_response(message, status=500)
+
         try:
-            dat = self._geocode(request.data)
-            # TODO enrichted data has to be written into database
-            return json_success(json.dumps(dat))
+            data = self._geocode(request.data)
```

Comment:
This method should use the data loaded from the table instead of the request-data if I'm not mistaken